olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -61,89 +118,137 @@ void OlpClient::SetSettings(const OlpClientSettings& settings) {
+  network::NetworkRequest::HttpVerb http_verb =
       network::NetworkRequest::HttpVerb::GET;
-  auto reqMethod = method;
-  if (reqMethod.compare("GET") == 0)
-    httpVerb = network::NetworkRequest::HttpVerb::GET;
-  else if (reqMethod.compare("PUT") == 0)
-    httpVerb = network::NetworkRequest::HttpVerb::PUT;
-  else if (reqMethod.compare("POST") == 0)
-    httpVerb = network::NetworkRequest::HttpVerb::POST;
-  else if (reqMethod.compare("DELETE") == 0)
-    httpVerb = network::NetworkRequest::HttpVerb::DEL;
+  auto request_method = method;
+  if (request_method.compare("GET") == 0)
+    http_verb = network::NetworkRequest::HttpVerb::GET;
+  else if (request_method.compare("PUT") == 0)
+    http_verb = network::NetworkRequest::HttpVerb::PUT;
+  else if (request_method.compare("POST") == 0)
+    http_verb = network::NetworkRequest::HttpVerb::POST;
+  else if (request_method.compare("DELETE") == 0)
+    http_verb = network::NetworkRequest::HttpVerb::DEL;
 
   std::string uri = base_url_ + path;
-  if (!queryParams.empty()) {
-    std::string queryParamsString = "?";
+  if (!query_params.empty()) {
+    std::string query_params_string = "?";
     auto first = true;
-    for (auto queryParam : queryParams) {
+    for (auto queryParam : query_params) {
       if (!first) {
-        queryParamsString.append("&");
+        query_params_string.append("&");
       }
       first = false;
-      queryParamsString.append(queryParam.first)
+      query_params_string.append(queryParam.first)
           .append("=")
           .append(queryParam.second);
     }
-    uri.append(queryParamsString);
+    uri.append(query_params_string);
+  }
+  auto network_request = std::make_shared<network::NetworkRequest>(
+      uri, 0, network::NetworkRequest::PriorityDefault, http_verb);
+
+  if (settings_.authentication_settings) {
+    network_request->AddHeader(
+        "Authorization",
+        "Bearer " + settings_.authentication_settings.get().provider());
   }
-  auto networkRequest = std::make_shared<network::NetworkRequest>(
-      uri, 0, network::NetworkRequest::PriorityDefault, httpVerb);
+
+  for (const auto& header : default_headers_) {
+    network_request->AddHeader(header.first, header.second);
+  }
+  for (const auto& header : header_params) {
+    network_request->AddHeader(header.first, header.second);
+  }
+
+  if (!content_type.empty()) {
+    network_request->AddHeader("Content-Type", content_type);
+  }
+
+  network_request->SetContent(post_body);
+
+  return network_request;
+}
+
+std::shared_ptr<http::NetworkRequest> OlpClient::CreateNewNetworkRequest(
+    const std::string& path, const std::string& method,
+    const std::multimap<std::string, std::string>& query_params,
+    const std::multimap<std::string, std::string>& header_params,
+    const std::shared_ptr<std::vector<unsigned char>>& post_body,
+    const std::string& content_type) const {
+  http::NetworkRequest::HttpVerb verb;
+  if (method == "GET") {
```

Comment:
A free function would suite better instead of duplication of the code at ln 128.